modules/setting/history/breaking_changes.go

```diff
@@ -0,0 +1,72 @@
+// Copyright 2023 The Gitea Authors. All rights reserved.
+// SPDX-License-Identifier: MIT
+
+package history
+
+import "code.gitea.io/gitea/modules/log"
+
+// This file is the only file that should be changed frequently in this package
+
+var currentGiteaVersion = getVersion("1.19")
+
+// Adds all previously removed settings
+// It should declare all breaking configuration changes in chronological order to ensure a monotone increasing error log
+func init() {
```

Comment:
So these information will be displayed before setting.Init ?

Reply:
Hmm, `init` seems uncontrollable. Is the logging system ready at the moment?

Reply:
> Is the logging system ready at the moment

Yes, that's one of the reasons it is in its own package. (The other is a clear separation of what belongs to what)
The package is only loaded when `PrintRemovedSettings` is called, which is exactly after the logging has been set up, and before any other processing has happened.
If we decide to drop these two trace calls, it doesn't even matter for the `init` method if the other package has been loaded already or not, the log is then only needed for `PrintRemovedSettings`.
So, what is the preferred course of action here?
Dropping the `trace` calls, or keeping them in and relying on the logger being initialized?
Or replacing the `init` with an `Init` we call ourselves?

> So these information will be displayed before setting.Init ?

No, it is called as soon as `setting.CfgProvider` has been set, but before any other settings have been initialized.
Since there is no `setting.Init` (anymore?), that's the best course of action.

Reply:
> Or replacing the `init` with an `Init` we call ourselves?

I think we should make the `Init` called explicitly, it makes the calling order controllable.